arch/arm/toolchain.mk

```diff
@@ -105,6 +105,13 @@ endif
 ifeq ($(ARM_CPU),arm1176jzf-s)
 ARCH_arm_COMPILEFLAGS += -mcpu=$(ARM_CPU)
 endif
+ifeq ($(ARM_CPU),cortex-r4f)
+ARCH_arm_COMPILEFLAGS += -march=armv7-r
+ARCH_arm_COMPILEFLAGS += -mcpu=$(ARM_CPU)
+ARCH_arm_COMPILEFLAGS += -mbig-endian
```

Comment:
may eventually want to break the big endian switch into a separate thing, but for now this will work since there are no other R class targets.